### PR TITLE
Dry

### LINKED ISSUE #3: Methods not being tested in Activity Class
Line 7 - findUser()
Line 16 - returnWeekOfData(week, userData)
Line 23 - returnWeek(week)  **I cannot find anywhere that this method is actually being invoked.  Do we need it?

These methods are not being tested directly.  The fact that they are used in other methods and those methods are passing tests indicates that they are working but testing them directly would be helpful in terms of someone being able to look at our test suite as a guide to our code.

### PR DESCRIPTION
**Pull request by hfaerber**
@Dunn-Austen @NaomiKC92 

#### What's this PR do?

This PR resolves issue (already closed) #13.  

It also would close another issue, #5 because it refactors the methods dynamically but we need to discuss whether we want the methods that return data for one week to be able to accept a start date for that week.   Currently these are build to accept a number (like 1) for the most recent 7 days or 2 for the 7 days before that. 

This PR will close issue #6 by correcly renaming the method to find the all time stair record. 

This PR removes the unused returnWeek() method in the Activity class that is noted in issue #3 and writes tests for the other two methods noted in issue #3.   This will fix #3. 

#### Where should the reviewer start?

Reviewers should start with the activity file as most changes started there.  All have been updated accordingly in the script file. 

#### How should this be manually tested?
Reviewers can use `npm test` to confirm that all tests are still passing and can use `npm run start` to confirm that all DOM manipulation is still in tact.

#### Any background context you want to provide?

Our next step will likely be determining where methods used by multiple classes should live.  
I'd love your input on this @Dunn-Austen and @NaomiKC92.   
The findUser() method in each class file is actually finding all the related activity/sleep/hydration stats for the specified user, not the user object itself.  I feel like this method and other shared methods might be best suited for a Utility class/file.   Khalid had mentioned one in my fit lit eval and I think it would make more sense than putting everything in User - but that could work too.  Thoughts?

#### What are the relevant tickets?

#3, #5 and #13 are relevant tickets.

#### Questions:

- Is there a blog post?  N/A
- Does the knowledge base need an update? N/A
- Does this add new dependencies which need to be added?  No

### CODE REVIEW

**Comment by hfaerber** [comment]
> Nice! I was hoping to implement the nested 'describe' block in this project. Good looks there! Regarding the lengthy tests where our expected results list out large series of objects (line 32 in the activity-test file has, for example, 15 objects spelled out in the test code), is keeping concise a consideration here? If that matters then maybe we could test the length of the expected array plus a property value at a specific index as a shorter alternative? I genuinely don't know so I thought I'd mention it. Please don't hate me.
> 
> Much appreciation for the Sunday evening work!
> -AD

That's a good question!  It definitely takes up a fair amount of space.  The main reason I built the additional tests is to be able to have a guide of what those methods are returning as I navigate the code base.  

Testing-wise, we could get away without having a specific test for them since they are essentially being tested in the tests for methods where they are being invoke within other methods.  

Maybe we could keep them lengthy for now, then once we figure out how we are reorganizing the code (if we're adding a class to utilize inheritance, etc), we could cut them down or cut them out altogether. 

What do you think @NaomiKC92 ?

**Comment by Dunn-Austen** [comment]
Hey Heather!

Your detailed messages make things real smooth for me. Love it. 

**- Re: Master / "Utility" class**
I don't have any experience with inheritance, so I'd have to research how best to implement it before being able to feel informed with a response. Or do we even need it? Maybe Naomi will have more to offer this the conversation.

**Re: Dynamic weeks using with a start date parameter**
I have a perhaps too-primitive approach to getting whatever week of data we want based off a passed start date whether for a single user or all of them. We'd probably want to look into something more optimal, but it's possible to get what we want just by counting indices. Our dataset's default sorting has laid out so that the data begins with user 1 on day 1 and then goes through each user consecutively up through user 50 on that same date, the marches forward again and again in the same order starting with user 1 for each date possible in the set. Fifty users, each with seven dates means that every 350 consecutive objects represents a week's worth of data. So we'd pass the start date and as long as we're filtering for that date AND user 1's id, we can slice that index through that value plus the next 349 indices to get a dynamic week date. This would only fail if the start date we pass through the function ends up being at the end of the data set where the remainder of days is less than what would comprise a week.

I've listed out some sample code for this approach below:

```
function findDataForSpecificWeekByStartDate(startDate) {
    let indexForUserOneAtSpecificDate = SLEEPDATA.findIndex(dataObject => {
          return (dataObject.userID === 1 && dataObject.date === date)
    });

    let allUserSleepDataForSpecificWeek = t t 
         SLEEPDATA.slice(indexForUserOneAtSpecificDate, (indexForUserOneAtSpecificDate + 349));
```

**Comment by hfaerber** [comment]
That was an incredibly articulate breakdown of how we could do this @Dunn-Austen!   Especially after reading your thoughts and seeing your code snippet, I think we could refactor to make these week-long methods more dynamic to improve the user experience.  

The most recent week could be the default for the dashboard display but if the user could choose from a drop-down calendar to update it, that could be a great UI/UX enhancement.   

I'd like to see us add a property to sleep/activity/hydration's constructors to automatically separate the stats for just the specified user and same them as a property that could be referenced in the other methods.    Alternatively we could invoke a method to do this in each class's 'Repo' equivalent and only pass the stats relevant to the specific user into each class in the first place.  This option might better meet the SRP.

In either case, that would make isolating any given week of data easier because the data set would already only have one user's stats to filter through.

**Comment by hfaerber** [comment]
@brittanystoroz @khalidwilliams

File: src/ActivityRepo.js [hunk 1]
```diff
@@ -13,11 +13,11 @@ class ActivityRepo {
...
   }
 
-  returnMostActive() {
-    let person = [...this.activityData].sort((a, b) => b.minutesActive - a.minutesActive)[0].userID;
-    let minActive = [...this.activityData].sort((a, b) => b.minutesActive - a.minutesActive)[0].minutesActive;
-    return [this.userData.find(user => user.id === person).name, minActive];
+  returnMostActive(metric) {
+    let person = [...this.activityData].sort((a, b) => b[metric] - a[metric])[0].userID;
+    let activity = [...this.activityData].sort((a, b) => b[metric] - a[metric])[0][metric];
+    return [this.userData.find(user => user.id === person).name, activity];
```
Comment (brittanystoroz) [inline src/ActivityRepo.js]: Beautiful refactoring!

File: test/ActivityRepo-test.js [hunk 2]
```diff
@@ -31,8 +31,20 @@ describe('ActivityRepo', () => {
...
+
+      it('should return the highest number of steps of all time', () => {
+        expect(activityRepo.returnMostActive('numSteps')).to.eql(['Mae Connelly', 14547]);
+      });
+
+      it('should return the highest number of stairs of all time', () => {
+        expect(activityRepo.returnMostActive('flightsOfStairs')).to.eql(['Herminia Witting', 46]);
+      });
+
+    })
```
Comment (brittanystoroz) [inline test/ActivityRepo-test.js]: Nice use of a nested describe block!

**Pull request merged**